src/main/java/com/minecolonies/core/colony/jobs/AbstractJobCrafter.java

```diff
@@ -51,6 +57,12 @@ public abstract class AbstractJobCrafter<AI extends AbstractEntityAIBasic<J, ? e
      */
     private int progress = 0;
 
+    /**
+     * The current map of secondary outputs that need to return to the warehouse after crafting is done.
+     */
+    @NotNull
+    private final Map<ItemStorage, Integer> secondaryOutputs = new HashMap<>();
```

Comment:
a Set<ItemStorage> should suffice, as item storage has an internal amount integer

Reply:
I figured with a Set I couldn't really make use of the `.compute` logic that a map has. It'd be slightly more annoying to have to deal with merging items together into 1 stack.

Reply:
My sweet Ray, sets have no getters, that's the whole problem

Reply:
I thought hashset had some kind of get by hash, I'm disappointed

Reply:
Then make it an object2Int map I would say, those are even more efficient for this particular usecase then.

Reply:
or you remove the existing from the set, addd the count to the new one and add it to the set again